elasticai_api/pytorch/controller.py

```diff
@@ -131,11 +134,13 @@ def train_one_batch_with_retries(self, func, *args, **kwargs):
             except RuntimeError:
                 traceback.print_exc()
                 self.restore()
+        if not allreduce_success:
+            raise RuntimeError("Failed to perform allreduce.")
```

Comment:
So the worker pod will fail after this runtime error?

Reply:
Yes. The worker will fail if it fails to retry to perform the allreduce operation.